### PR TITLE
chore: add cSpell for spell checking

### PR DESCRIPTION
**Pull request by nschonni**
## Motivation

Used `cspell --no-progress --words-only --unique "**" > project-words.txt` to create a base dictionary after ignoring some of the files that added in a bunch of one-off words in the users.txt and the translations

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -22,6 +22,8 @@ jobs:
       - name: Check immutable yarn.lock
         run: git diff --exit-code
       - name: Lint
-        run: yarn lint:ci
+        run: |
+          echo "::add-matcher::.github/workflows/cspell-problem-matcher.json"
```
Comment (Josh-Cena) [inline .github/workflows/lint.yml:26]: As an aside, we haven't seen eslint error matchers for a long time. Maybe that's a problem with our setup-node config?

File: project-words.txt [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+noicon
+noindex
+nojekyll
+noninteractive
+noreply
+npmjs
+nprogress
+NTFS
+Nuxt
+OAGI
```
Comment (Josh-Cena) [inline project-words.txt:254]: There are a lot of these random 4-letter words. Maybe we need to ignore some other folder? Where are they coming from?

Comment (nschonni) [reply project-words.txt:228]: Found some in a `.map` file and excluded those, along with removing sum upper/lowercase dups

File: .cspell.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "version": "0.2",
+  "gitignoreRoot": ".",
+  "dictionaries": [
+    "css",
```
Comment (armano2) [inline .cspell.json:5]: can you add `filetypes` and `npm` and `misc`, that should limit number of words in custom dictionary by alot

Comment (nschonni) [reply .cspell.json:5]: Tried adding those locally, but there any additional words it removed. I think it was mostly covered by excluding the DOCX and XYZ file types

File: .cspell.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "package.json",
+    "**/yarn.lock",
+    "yarn.lock",
+    "project-words.txt",
+    "website/src/data/users.tsx",
+    "**/*.map",
+    "**/*.xyz",
+    "**/*.docx"
+  ],
+  "ignoreRegExpList": ["Email", "Urls", "{#\\w*}"],
```
Comment (Josh-Cena) [inline .cspell.json:37]:
```suggestion
  "ignoreRegExpList": ["Email", "Urls", "{#[\\w-]*}"],
```

By my understanding, `\w` doesn't match dashes, right? But anchors can contain dashes?

**Review by Josh-Cena — APPROVED** [review]
This is starting to look great! Huge anticipations for a much more polished codebase in the near future

File: packages/docusaurus-plugin-content-docs/src/slug.ts [hunk 5]
```diff
@@ -21,14 +21,14 @@ import type {NumberPrefixParser} from '@docusaurus/plugin-content-docs';
 
 export default function getSlug({
   baseID,
-  frontmatterSlug,
+  frontMatterSlug,
```
Comment (nschonni) [inline packages/docusaurus-plugin-content-docs/src/slug.ts:24]: Is this a breaking change?

Comment (Josh-Cena) [reply packages/docusaurus-plugin-content-docs/src/slug.ts:24]: No, we don't expose this function to the end users.

**Pull request merged**